src/main/java/org/sonarlint/intellij/ui/AbstractIssuesPanel.java

```diff
@@ -21,35 +21,64 @@
 
 import com.intellij.ide.OccurenceNavigator;
 import com.intellij.ide.util.PropertiesComponent;
+import com.intellij.openapi.actionSystem.ActionGroup;
+import com.intellij.openapi.actionSystem.ActionManager;
+import com.intellij.openapi.actionSystem.ActionToolbar;
+import com.intellij.openapi.components.ServiceManager;
 import com.intellij.openapi.editor.RangeMarker;
 import com.intellij.openapi.fileEditor.OpenFileDescriptor;
 import com.intellij.openapi.project.Project;
 import com.intellij.openapi.ui.SimpleToolWindowPanel;
 import com.intellij.openapi.ui.Splitter;
 import com.intellij.ui.treeStructure.Tree;
+import java.awt.event.KeyAdapter;
+import java.awt.event.KeyEvent;
+import java.util.Collections;
 import javax.annotation.CheckForNull;
 import javax.annotation.Nullable;
+import javax.swing.Box;
 import javax.swing.JComponent;
 import javax.swing.tree.DefaultMutableTreeNode;
+import javax.swing.tree.DefaultTreeModel;
 import javax.swing.tree.TreeNode;
 import javax.swing.tree.TreePath;
+import org.sonarlint.intellij.editor.SonarLintHighlighting;
+import org.sonarlint.intellij.issue.LiveIssue;
 import org.sonarlint.intellij.ui.nodes.AbstractNode;
 import org.sonarlint.intellij.ui.nodes.IssueNode;
-import org.sonarlint.intellij.ui.tree.TreeModelBuilder;
+import org.sonarlint.intellij.ui.nodes.LocationNode;
+import org.sonarlint.intellij.ui.tree.FlowsTree;
+import org.sonarlint.intellij.ui.tree.FlowsTreeModelBuilder;
+import org.sonarlint.intellij.ui.tree.IssueTree;
+import org.sonarlint.intellij.ui.tree.IssueTreeModelBuilder;
 
 abstract class AbstractIssuesPanel extends SimpleToolWindowPanel implements OccurenceNavigator {
+  private static final String ID = "SonarLint";
   protected Project project;
+  protected SonarLintHighlighting highlighting;
+  protected SonarLintRulePanel rulePanel;
   protected Tree tree;
-  protected TreeModelBuilder treeBuilder;
+  protected IssueTreeModelBuilder treeBuilder;
+  protected FlowsTree flowsTree;
+  protected FlowsTreeModelBuilder flowsTreeBuilder;
+  protected ActionToolbar mainToolbar;
 
-  AbstractIssuesPanel() {
+  AbstractIssuesPanel(Project project) {
     super(false, true);
+    this.project = project;
+    this.highlighting = ServiceManager.getService(project, SonarLintHighlighting.class);
+
+    addToolbar();
+    createFlowsTree();
+    createIssuesTree();
   }
 
-  protected JComponent createSplitter(JComponent c1, JComponent c2, String proportionProperty) {
-    float savedProportion = PropertiesComponent.getInstance(project).getFloat(proportionProperty, 0.65f);
+  abstract protected String getToolbarGroupId();
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Reorder the modifiers to comply with the Java Language Specification. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AModifiersOrderCheck)